src/components/PostDetails.tsx

```diff
@@ -2,105 +1,137 @@
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+export const PostDetails: React.FC<Props> = ({
+  currPost,
+  modalWindow,
+  setModalWindow,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [btnLoader, setBtnLoader] = useState(false);
+  const [commentLoader, setCommentLoader] = useState(false);
 
-        <div className="block">
-          <Loader />
+  const onAdd = ({ name, email, body }: CommentData) => {
+    setBtnLoader(true);
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+    const postId = currPost?.id || 0;
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+    return addComment({ postId, name, email, body })
+      .then(newComment => setComments([...comments, newComment]))
+      .catch(err => {
+        setError(`Failed to add comment: ${err.message || 'Unknown error'}`);
+      })
+      .finally(() => setBtnLoader(false));
+  };
 
-          <p className="title is-4">Comments:</p>
+  const onDelete = (commentId: number) => {
+    setComments(currComments => {
+      return currComments.filter(comment => comment.id !== commentId);
+    });
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+    deleteComment(commentId)
+      .then(() => {})
+      .catch(err => {
+        setError(`Failed to delete comment: ${err.message || 'Unknown error'}`);
+      });
+  };
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+  useEffect(() => {
+    if (currPost) {
+      setCommentLoader(true);
+      getComments(currPost.id)
+        .then(setComments)
+        .catch(err => {
+          setError(
+            `Failed to fetch comments: ${err.message || 'Unknown error'}`,
+          );
+        })
+        .finally(() => setCommentLoader(false));
+    }
+  }, [currPost, setComments]);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="content" data-cy="PostDetails">
```

Comment:
The `data-cy` attribute is duplicated for the `PostDetails` div. Ensure that each `data-cy` attribute is unique to avoid confusion in testing and debugging.